### PR TITLE
Add CLI support to generate morphometrics file.

### LINKED ISSUE #421: Add CLI to compute morphometrics
We should implement a CLI to compute morphometrics and output a spreadsheet, as it is done in fsleyes.
This would be helpful for users who are not using fsleyes, and easier than dealing with the notebook.

### PR DESCRIPTION
**Pull request by vs74**
Following discussions in issue #421, this PR aims to implement CLI to generate morphometrics file.

**CLI COMMAND :** `axondeepseg_morphometrics -i path_to_image.png  -f  axon_morphometrics.xlsx -s pixel_size`  

**DOCUMENTATION** - https://axondeepseg--434.org.readthedocs.build/en/434/documentation.html#morphometrics

Fixes #421

**Test the PR** 
Kindly refer to the [RTD](https://axondeepseg--434.org.readthedocs.build/en/434/documentation.html#morphometrics) and test the functionalities as mentioned under the **Morphometrics** section of the RTD.

**DONE: -**

- [x] Documentation: Add an example regarding usage for CLI
- [x] Feature: Should generate morphometrics from directory different from image directory
- [x] Feature: Morphometrics of images from multiple folders much like `axondeepseg` CLI.
- [x] Documentation: Morphometrics of images from multiple folders much like `axondeepseg` CLI.
- [x] Add tests

### CODE REVIEW

**Comment by vs74** [comment]
Does anyone know if there is a way to render RTD from the forked repo and include in this PR so it would be easy to review this PR ?

**Comment by jcohenadad** [comment]
> Does anyone know if there is a way to render RTD from the forked repo and include in this PR so it would be easy to review this PR ?

link of the built RTD for the present PR is in the checks:
image

**Comment by vs74** [comment]
@mariehbourget, I've added some literature studies here, but these are not sufficient. If you happen to know literature where axon shape being circle or ellipse are used for morphometrics, feel free to contribute in the documentation of this PR or add them in the comments, which I will add on in the RTD. Thanks!!

**Comment by vs74** [comment]
This PR is ready for review.

File: AxonDeepSeg/morphometrics/compute_morphometrics.py [hunk 1]
```diff
@@ -154,33 +161,37 @@ def get_axon_morphometrics(im_axon, path_folder=None, im_myelin=None, pixel_size
 
     return stats_array
 
-def evaluate_myelin_thickness_in_px(axon_object, axonmyelin_object):
+def evaluate_myelin_thickness_in_px(axon_object, axonmyelin_object, axon_shape):
     """
-    Returns the equivalent thickness of a myelin ring around an axon of a
-    given equivalent diameter (see note [1] below). The result is in pixels.
+    Returns the  thickness of a myelin ring around an axon of a
+    given equivalent diameter (see note [1] below) or minimum axis diameter of an ellipse. The result is in pixels.
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/compute_morphometrics.py:167]:
```suggestion
    given equivalent diameter (see note [1] below) or minor axis of an ellipse. The result is in pixels.
```

File: AxonDeepSeg/morphometrics/compute_morphometrics.py [hunk 2]
```diff
@@ -154,33 +161,37 @@ def get_axon_morphometrics(im_axon, path_folder=None, im_myelin=None, pixel_size
...
     :param axon_object (skimage.measure._regionprops): object returned after
         measuring a axon labeled region
     :param axonmyelin_object (skimage.measure._regionprops): object returned after
         measuring a axon with myelin labeled region
-
+   :param axon_shape: str: shape of the axon, can either be either be circle or an ellipse
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/compute_morphometrics.py:172]:
```suggestion
   :param axon_shape: str: shape of the axon, can either be circle or ellipse
```

File: AxonDeepSeg/morphometrics/compute_morphometrics.py [hunk 3]
```diff
@@ -265,13 +275,14 @@ def load_axon_morphometrics(path_folder):
...
 
-def draw_axon_diameter(img, path_prediction, pred_axon, pred_myelin):
+def draw_axon_diameter(img, path_prediction, pred_axon, pred_myelin, axon_shape="circle"):
     """
     :param img: sample grayscale image (png)
     :param path_prediction: full path to the segmented file (*_seg-axonmyelin.png)
         from axondeepseg segmentation output
     :param pred_axon: axon mask from axondeepseg segmentation output
     :param pred_myelin: myelin mask from axondeepseg segmentation output
+    :param axon_shape: str: shape of the axon, can either be either be circle or an ellipse
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/compute_morphometrics.py:285]:
```suggestion
    :param axon_shape: str: shape of the axon, can either be circle or ellipse
```

File: AxonDeepSeg/morphometrics/compute_morphometrics.py [hunk 4]
```diff
@@ -324,11 +335,12 @@ def save_map_of_axon_diameters(path_folder, axon_diameter_figure):
     axon_diameter_figure.savefig(file_path)
 
 
-def get_aggregate_morphometrics(pred_axon, pred_myelin, path_folder):
+def get_aggregate_morphometrics(pred_axon, pred_myelin, path_folder, axon_shape="circle"):
     """
     :param pred_axon: axon mask from axondeepseg segmentation output
     :param pred_myelin: myelin mask from axondeepseg segmentation output
     :param path_folder: absolute path of folder containing pixel size file
+    :param axon_shape: str: shape of the axon, can either be either be circle or an ellipse
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/compute_morphometrics.py:343]:
```suggestion
    :param axon_shape: str: shape of the axon, can either be circle or ellipse
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
 
-def launch_morphometrics_computation(path_img, path_prediction):
+def launch_morphometrics_computation(path_img, path_prediction, axon_shape="cicle"):
     """
     This function is equivalent to the morphometrics_extraction notebook of AxonDeepSeg.
     It automatically performs all steps (computations, savings, displays,...) of the
     morphometrics extraction of a given sample.
     :param path_img: path of the input image (microscopy sample)
     :param path_prediction: path of the segmented image (output of AxonDeepSeg)
+    :param axon_shape: str: shape of the axon, can either be either be circle or an ellipse
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:26]:
```suggestion
    :param axon_shape: str: shape of the axon, can either be circle or ellipse
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 6]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
         write_aggregate_morphometrics(path_folder, aggregate_metrics)
+
+
+
+def main(argv=None):
+    ap = argparse.ArgumentParser(formatter_class=RawTextHelpFormatter)
+
+    
+    # Setting the arguments of the saving the morphometrics in excel file
+    ap.add_argument('-s', '--sizepixel', required=False, help='Pixel size of the image(s) to segment, in micrometers. \n'+
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:75]:
```suggestion
    ap.add_argument('-s', '--sizepixel', required=False, help='Pixel size of the image(s) to compute morphometrics, in micrometers. \n'+
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 7]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                                                              'If no pixel size is specified, a pixel_size_in_micrometer.txt \n'+
+                                                              'file needs to be added to the image folder path. The pixel size \n'+
+                                                              'in that file will be used for the segmentation.',
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:78]:
```suggestion
                                                              'in that file will be used for the morphometrics computation.',
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 8]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                                                              default=None)
+
+
+    ap.add_argument('-i', '--imgpath', required=True, nargs='+', help='Path to the image to be segmented ')
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:82]:
```suggestion
    ap.add_argument('-i', '--imgpath', required=True, nargs='+', help='Path to the image.')
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 9]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+
+    
+    ap.add_argument('-f', '--filename', required=False,  help='Name of the excel file in which the morphometrics file will be  stored',
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:85]:
```suggestion
    ap.add_argument('-f', '--filename', required=False,  help='Name of the excel file in which the morphometrics will be stored',
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 10]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+
+    for current_path_target in path_target_list:
+
+        if current_path_target.suffix.lower() in validExtensions:
+
+            #load the axon mask 
+            if (Path(str(current_path_target.with_suffix("")) + str(axon_suffix))).exists():
+                pred_axon = image.imread(str(current_path_target.with_suffix("")) + str(axon_suffix))
+            else: 
+                print("ERROR: Segmented axon mask is not present in the image folder  ",
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:118]:
```suggestion
                print("ERROR: Segmented axon mask is not present in the image folder. ",
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 11]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                                    "Please check that the axon mask is located in the image folder ",
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:119]:
```suggestion
                                    "Please check that the axon mask is located in the image folder. ",
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 12]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                                    "If it is not present, perform segmentation of the image first using ADS."
+                    )
+                sys.exit(3)
+
+            #load myelin mask    
+            if (Path(str(current_path_target.with_suffix("")) + str(myelin_suffix))).exists():
+                pred_myelin = image.imread(str(current_path_target.with_suffix("")) + str(myelin_suffix))
+            else: 
+                print("ERROR: Segmented myelin mask is not present in the image folder.  ",
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:128]:
```suggestion
                print("ERROR: Segmented myelin mask is not present in the image folder. ",
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 13]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                    )
+
+
+
+            # save the current contents in the file
+            if not (filename.lower().endswith((".xlsx", ".csv"))):  # If the user didn't add the extension, add it here
+                filename = filename + ".xlsx"
+            try:
+                # Export to excel
+                pd.DataFrame(x).to_excel(current_path_target.parent /  filename)
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:201]: There is no export to csv file. A `.csv` extension in `-f` will results in an error `ValueError: No engine for filetype: 'csv' `

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 14]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                    psm = float(resolution_file.read())
+                else:
+
+                    print("ERROR: No pixel size is provided, and there is no pixel_size_in_micrometer.txt file in image folder. ",
+                                    "Please provide a pixel size (using argument -s), or add a pixel_size_in_micrometer.txt file ",
+                                    "containing the pixel size value."
+                    )
+                    sys.exit(3)
+
+
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:152]: Empty lines to remove?

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 15]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                        stats['axon_diam'],
+                        stats['myelin_thickness'],
+                        stats['axonmyelin_area'],
+                        stats['solidity'],
+                        stats['eccentricity'],
+                        stats['orientation']
+                        )],
+                        dtype=x.dtype)
+                    )
+
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:193]: Empty lines to remove?

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 16]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+                print("The path(s) specified is/are not image(s). Please update the input path(s) and try again.")
+                break
+        
+    sys.exit(0)
+
+# Calling the script
+if __name__ == '__main__':
+    main()
+
+
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:216]:
```suggestion
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py [hunk 17]
```diff
@@ -43,15 +52,166 @@ def launch_morphometrics_computation(path_img, path_prediction):
...
+
```
Comment (mariehbourget) [inline AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:217]:
```suggestion
```

File: docs/source/documentation.rst [hunk 18]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+
+**Required arguments:**
+
+-i IMGPATH
+                    Path to the image file whose morphometrics needs to be calculated.
+
+**Optional arguments:**
+
+-s SIZEPIXEL        Pixel size of the image(s) to segment, in micrometers. 
+                    If no pixel size is specified, a **pixel_size_in_micrometer.txt** file needs to be added to the image folder path ( that file should contain a single float number corresponding to the resolution of the image, i.e. the pixel size). The pixel size in that file will be used for the segmentation.
```
Comment (mariehbourget) [inline docs/source/documentation.rst:338]:
```suggestion
                    If no pixel size is specified, a **pixel_size_in_micrometer.txt** file needs to be added to the image folder path (that file should contain a single float number corresponding to the resolution of the image, i.e. the pixel size). The pixel size in that file will be used for the morphometrics computation.
```

File: docs/source/documentation.rst [hunk 19]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+
+
+-a AXONSHAPE       Axon shape
+                    **circle:** Axon shape is considered as circle. In this case, diameter is computed using equivalent diameter. 
+                    **ellipse:** Axon shape is considered as an ellipse. In this case, diameter is computed using ellipse minor axis.
+                    The default axon shape is set to **circle**.
+
+-f FILENAME         Name of the excel file in which the morphometrics file will be stored.
```
Comment (mariehbourget) [inline docs/source/documentation.rst:346]:
```suggestion
-f FILENAME         Name of the excel file in which the morphometrics will be stored.
```

File: docs/source/documentation.rst [hunk 20]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+                    The excel file extension can either be **.xlsx** or **.csv**.
```
Comment (mariehbourget) [inline docs/source/documentation.rst:347]: The `csv` option is not working, see comment in `launch_morphometrics_computation.py`.

File: docs/source/documentation.rst [hunk 21]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+    ---- 77.png
+    ---- 77_seg-axon.png
+    ---- 77_seg-axonmyelin.png
+    ---- 77_seg-myelin.png
+    ---- pixel_size_in_micrometer.txt
+    ---- morphometrics.xlsx
+    ...
+
+.. NOTE :: By default, AxonDeepSeg treats axon shape as **circle** and the calculation of the diameter is based on the axon area of the mask. 
+           For each axons, the equivalent diameter is computed, which is the diameter of a circle with the same area as the axon ::
```
Comment (mariehbourget) [inline docs/source/documentation.rst:368]:
```suggestion
           For each axons, the equivalent diameter is computed, which is the diameter of a circle with the same area as the axon. ::
```

File: docs/source/documentation.rst [hunk 22]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+
+**Studies using Ellipse as axon shape:**
+
+* Payne et al: https://pubmed.ncbi.nlm.nih.gov/21381867/
+* Payne et al: https://pubmed.ncbi.nlm.nih.gov/22879411/
+* Fehily et al: https://pubmed.ncbi.nlm.nih.gov/30702755/
+
+
+.. NOTE :: In the literature, both equivalent diameter and ellipse minor axis are used to compute the morphometrics. 
+           Thus, depending on the usecase the user is advised to choose axon shape accordingly.
```
Comment (mariehbourget) [inline docs/source/documentation.rst:428]:
```suggestion
           Thus, depending on the usecase, the user is advised to choose axon shape accordingly.
```

File: notebooks/04-morphometrics_extraction.ipynb [hunk 23]
```diff
@@ -86,8 +89,11 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "axon_shape = True #String, circle: axon shape is considered circle for computing morphometrics\n",
```
Comment (mariehbourget) [inline notebooks/04-morphometrics_extraction.ipynb:92]:
```suggestion
    "axon_shape = "circle" #String, circle: axon shape is considered circle for computing morphometrics\n",
```

File: notebooks/04-morphometrics_extraction.ipynb [hunk 24]
```diff
@@ -86,8 +89,11 @@
...
+    "             #        ellipse: axon shape is considered ellipse for computing morphometrics\n",
```
Comment (mariehbourget) [inline notebooks/04-morphometrics_extraction.ipynb:93]:
```suggestion
                           # String, ellipse: axon shape is considered ellipse for computing morphometrics\n",
```

File: notebooks/04-morphometrics_extraction.ipynb [hunk 25]
```diff
@@ -187,7 +195,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "fig = draw_axon_diameter(img,path_pred,pred_axon,pred_myelin)\n",
+
+    "fig = draw_axon_diameter(img,path_pred,pred_axon,pred_myelin, axon_shape=shape)\n",
```
Comment (mariehbourget) [inline notebooks/04-morphometrics_extraction.ipynb:199]:
```suggestion
    "fig = draw_axon_diameter(img, path_pred, pred_axon, pred_myelin, axon_shape=shape)\n",
```

File: notebooks/04-morphometrics_extraction.ipynb [hunk 26]
```diff
@@ -207,7 +217,7 @@
    "outputs": [],
    "source": [
     "# Compute aggregate metrics\n",
-    "aggregate_metrics = get_aggregate_morphometrics(pred_axon,pred_myelin,path_folder)\n",
+    "aggregate_metrics = get_aggregate_morphometrics(pred_axon,pred_myelin,path_folder, shape=shape)\n",
```
Comment (mariehbourget) [inline notebooks/04-morphometrics_extraction.ipynb:220]:
```suggestion
    "aggregate_metrics = get_aggregate_morphometrics(pred_axon, pred_myelin, path_folder, shape=shape)\n",
```

File: test/morphometrics/test_compute_morphometrics.py [hunk 27]
```diff
@@ -228,6 +337,20 @@ def test_draw_axon_diameter_creates_file_in_expected_location(self):
 
         assert result_path.is_file()
         result_path.unlink()
+    
+    @pytest.mark.unit
+    def test_draw_axon_diameter_creates_file_in_expected_location_with_axon_as_ellipse(self):
+        img = imageio_imread(self.test_folder_path / 'image.png')
+        path_prediction = self.test_folder_path / 'AxonDeepSeg_seg-axonmyelin.png'
```
Comment (mariehbourget) [inline test/morphometrics/test_compute_morphometrics.py:344]: Should we use the standard axonmyelin_suffix here for the filename?

File: docs/source/documentation.rst [hunk 28]
```diff
@@ -307,12 +310,124 @@ Then, use the following command::
...
+           If you wish to treat axon shape as an ellipse, you can set the  **-a** argument to be **ellipse**.
+           When axon shape is set to ellipse, the calculation of the diameter is based on ellipse minor axis::
+            
+            axondeepseg -i test_segmentation/test_sem_image/image1_sem/77.png -a ellipse
+
+Morphometrics of images from multiple folders
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To generate morphometrics of images which are located in different folders, specify the path of the image folders using the **-i** argumument of the CLI separated by space. For instance, to compute morphometrics of the images present in the folders **'test_sem_image/image1_sem/'** and **'test_sem_image/image2_sem/'** of the test dataset, use the following command::
+
+    axondeepseg_morphometrics -i test_sem_image/image1_sem/77.png test_sem_image/image2_sem/image.png
```
Comment (mariehbourget) [inline docs/source/documentation.rst:379]:
```suggestion
    axondeepseg_morphometrics -i test_segmentation/test_sem_image/image1_sem/77.png test_segmentation/test_sem_image/image2_sem/image.png
```

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py (hunk 13 above)
Comment (vs74) [reply AxonDeepSeg/morphometrics/launch_morphometrics_computation.py:201]: Good catch, I'll update the code and tests for the same.

File: test/morphometrics/test_compute_morphometrics.py (hunk 27 above)
Comment (vs74) [reply test/morphometrics/test_compute_morphometrics.py:344]: yes, I've fixed it.

File: AxonDeepSeg/morphometrics/launch_morphometrics_computation.py (hunk 13 above)
Comment (vs74) [reply AxonDeepSeg/morphometrics/launch_morphometrics_computation.py]: Added test for saving filename with `.csv` extension and fixed that error.

File: docs/source/documentation.rst (hunk 20 above)
Comment (vs74) [reply docs/source/documentation.rst:347]: Fixed this.

**Comment by vs74** [comment]
Thanks @mariehbourget for the review of this PR. I've addressed your review suggestions. In addition, as discussed in the last meeting I've changed the default name of the morphometrics file from `Morphometrics` to `axon_morphometrics`.

**Review by mariehbourget — COMMENTED** [review]
I tested again with the new changes. Everything is working as expected including new csv option.
However, I could not open the notebook 04 with the following error
```
Error loading notebook
Unreadable Notebook: /home/mhbourget/axondeepseg/notebooks/04-morphometrics_extraction.ipynb NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells": [\\n {\\n "cell_type": "m...')
```
I think this is not related to this PR but to #399 that should be merged first.

**Comment by mathieuboudreau** [comment]
What’s the current statut for that PR @mariehbourget @vs74 ? Is merging #399 essential before completing this feature?

File: docs/source/documentation.rst [hunk 29]
```diff
@@ -269,12 +272,81 @@ Then, use the following command::
...
     axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/ test_segmentation/test_sem_image/image2_sem/
 
+
+Morphometrics
+---------------
+
+You can generate morphometrics using AxonDeepSeg via the command line interface.
+
+Syntax
+~~~~~
```
Comment (mariehbourget) [inline docs/source/documentation.rst:288]:
```suggestion
~~~~~~
```

File: docs/source/documentation.rst [hunk 30]
```diff
@@ -269,12 +272,81 @@ Then, use the following command::
...
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 To segment images that are located in different folders, specify the path to the folders in the **-i** argument, one after the other. For instance, to segment all the images of folders **'test_sem_image/image1_sem/'** and **'test_sem_image/image2_sem/'** of the test dataset, use the following command::
 
     axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/ test_segmentation/test_sem_image/image2_sem/
 
+
+Morphometrics
+---------------
```
Comment (mariehbourget) [inline docs/source/documentation.rst:283]:
```suggestion
-------------
```

**Review by mariehbourget — COMMENTED** [review]
I tested the morphometrics CLI and review the documentation, everything is working as expected for 1 file of multiple files each in a different folder.

One question though, should we extend the functionality to multiple files in a single folder?
It would make sense since our CLI for segmentation has this option. But we would need a mechanism to avoid overwriting the xls file with each image.

**Comment by vs74** [comment]
Thanks, @mariehbourget for testing this PR. 

> One question though, should we extend the functionality to multiple files in a single folder?
> It would make sense since our CLI for segmentation has this option. But we would need a mechanism to avoid overwriting the xls file with each image.

Thanks for pointing out the possibility of the functionality of generating multiple files in a single folder I also gave a thought about it while implementing this PR, but since it required overwriting the excel files, I didn't implement it. 

Right now, I've asked the user about this [here](https://github.com/neuropoly/axondeepseg/discussions/505). Let's see what his response is. 

If the user, doesn't require generating multiple files in a single folder, then we can merge this PR and open a new issue about the additional functionality. For this, we've to fix the renaming condition of the excel files generated. For instance, an excel file could be renamed as 'image_name_axon_morphometrics.xlsx'. This would meet the user's urgent requirement. 

On the contrary, if he requires generating multiple files in a single folder, it will take 1-2 days for this PR to be ready as renaming of the morphometrics file and some tests need to added in this PR.

**Comment by vs74** [comment]
Looks like the user has images in different directories ( see [here](https://github.com/neuropoly/axondeepseg/discussions/505#discussioncomment-660850)). I say for now, we can merge this PR If it is alright.

I'll open a follow-up issue on adding support for generating morphometrics when all the images are within a single directory.

**Comment by mariehbourget** [comment]
> Looks like the user has images in different directories ( see [here](https://github.com/neuropoly/axondeepseg/discussions/505#discussioncomment-660850)). I say for now, we can merge this PR if it is alright.
> 
> I'll open a follow-up issue on adding support for generating morphometrics when all the images are within a single directory.
> 
> Let me know what do you think @mariehbourget.

Yes, that is ok with me, thanks! I'll approve the PR and let you do the merge.

**Review by mariehbourget — APPROVED** [review]

**Pull request merged**